pkg/core/interop_neo.go

```diff
@@ -21,22 +30,63 @@ func storageFind(ic *interop.Context) error {
 		return fmt.Errorf("%T is not a StorageContext", stcInterface)
 	}
 	prefix := ic.VM.Estack().Pop().Bytes()
+	opts := ic.VM.Estack().Pop().BigInt().Int64()
+	if opts&FindKeysOnly != 0 && opts&FindValuesOnly != 0 {
+		return fmt.Errorf("%w: KeysOnly conflicts with ValuesOnly", errFindInvalidOptions)
+	}
 	siMap, err := ic.DAO.GetStorageItemsWithPrefix(stc.ID, prefix)
 	if err != nil {
 		return err
 	}
 
 	filteredMap := stackitem.NewMap()
 	for k, v := range siMap {
-		filteredMap.Add(stackitem.NewByteArray(append(prefix, []byte(k)...)), stackitem.NewByteArray(v.Value))
+		var key []byte
+		if opts&FindRemovePrefix == 0 {
+			key = append(prefix, []byte(k)...)
+		} else {
+			key = []byte(k)
+		}
```

Comment:
Not exactly the way it is done in C# code. Here `prefix` is an array of bytes, and in C# it is just a single byte, see https://github.com/neo-project/neo/pull/2190/files#diff-2bbcf2b5eefab34a31011387ebf58e6a251ca395d800376cf6ef22bc5039475eR36-R37 .

Reply:
Fixed.